crates/bevy_solari/src/pathtracer/pathtracer.wgsl

```diff
@@ -77,3 +92,60 @@ fn pathtrace(@builtin(global_invocation_id) global_id: vec3<u32>) {
     textureStore(accumulation_texture, global_id.xy, vec4(new_color, old_color.a + 1.0));
     textureStore(view_output, global_id.xy, vec4(new_color, 1.0));
 }
+
+struct NextBounce {
+    wi: vec3<f32>,
+    pdf: f32,
+    perfectly_specular_bounce: bool,
+}
+
+fn importance_sample_next_bounce(wo: vec3<f32>, ray_hit: ResolvedRayHitFull, rng: ptr<function, u32>) -> NextBounce {
+    let is_perfectly_specular = ray_hit.material.roughness < 0.0001 && ray_hit.material.metallic > 0.9999;
+    if is_perfectly_specular {
+        return NextBounce(reflect(-wo, ray_hit.world_normal), 1.0, true);
+    }
+    let diffuse_weight = mix(mix(0.4f, 0.9f, ray_hit.material.perceptual_roughness), 0.f, ray_hit.material.metallic);
```

Comment:
whats the 0.4 and 0.9 about? and why the mix(mix(? isnt the outer mix really just a multiply by (1.0-metallic) since its a mix towards 0

Reply:
Just random numbers that @SparkyPotato found to work well iirc.